### PR TITLE
[docs][tune] Make search algorithm, scheduler docs better!

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?
This PR revises the search algorithm documentation to be more
comprehensive and informative.

TODO:

 - Move content into the API documentation
 - P1: Show common search spaces
 - Add a matrix for what things are compatible.
 - Add a guide for how to optimize hyperparameters/use a trial
scheduler. ([P] Tutorial: Guide to HP Optimization)

## Related issue number
Waiting for Refactor Search PR to close.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: doc/source/tune/api_docs/schedulers.rst [hunk 1]
```diff
@@ -1,38 +1,173 @@
...
+     - Yes
+     - `Link <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/hyperband_example.py>`__
+   * - :ref:`BOHB <tune-scheduler-bohb>`
+     - Yes
+     - Only TuneBOHB
+     - `Link <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/bohb_example.py>`__
+   * - :ref:`Population Based Training <tune-scheduler-pbt>`
+     - Yes
+     - Not Compatible
+     - `Link <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/pbt_example.py>`__
```
Comment (richardliaw) [inline doc/source/tune/api_docs/schedulers.rst:52]: @drozzy is this helpful?

File: doc/source/tune/api_docs/suggestion.rst [hunk 2]
```diff
@@ -1,73 +1,195 @@
-.. _searchalg-ref:
+.. _tune-search-alg:
 
 Search Algorithms (tune.suggest)
 ================================
 
-.. _repeater-doc:
+Tune's Search Algorithms are wrappers around open-source optimization libraries for efficient hyperparameter selection. These integrations optimize the given trainable over a provided search space. You can utilize these search algorithms as follows:
+
```
Comment (anabranch) [inline doc/source/tune/api_docs/suggestion.rst:7]: can you give a warning that each one is implemented differently or whatever we chatted about earlier today?

File: doc/source/tune/api_docs/schedulers.rst [hunk 3]
```diff
@@ -1,38 +1,173 @@
-.. _schedulers-ref:
+.. _tune-schedulers:
 
 Trial Schedulers (tune.schedulers)
 ==================================
 
-FIFOScheduler
-~~~~~~~~~~~~~
+In Tune, some hyperparameter optimization algorithms are written as "scheduling algorithms". These Trial Schedulers can early terminate bad trials, pause trials, clone trials, and alter hyperparameters of a running trial.
```
Comment (anabranch) [inline doc/source/tune/api_docs/schedulers.rst:6]: Is there a recommended trial scheduler? Is there one that people should always use?

**Review by bllchmbrs — DISMISSED** [review]

File: doc/source/tune/_tutorials/tune-usage.rst [hunk 4]
```diff
@@ -282,8 +282,11 @@ Note that in the above example the currently running trials will not stop immedi
 Logging/Tensorboard
 -------------------
 
+Tune by default will log results for Tensorboard, CSV, and JSON formats. If you need to log something lower level like model weights or gradients, see :ref:`Tighter Logging <tighter-logging>`.
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/tune/_tutorials/tune-usage.rst]: -> trainable logging

File: doc/source/tune/api_docs/logging.rst [hunk 5]
```diff
@@ -3,10 +3,95 @@
...
+Custom Loggers
+--------------
+
+You can pass in your own logging mechanisms to output logs in custom formats as follows:
+
+.. code-block:: python
+
+    from ray.tune.logger import DEFAULT_LOGGERS
+
+    tune.run(
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/tune/api_docs/logging.rst:53]: done

File: doc/source/tune/api_docs/logging.rst [hunk 6]
```diff
@@ -3,10 +3,95 @@
...
+        MyTrainableClass,
+        name="experiment_name",
+        loggers=DEFAULT_LOGGERS + (CustomLogger1, CustomLogger2)
+    )
+
+These loggers will be called along with the default Tune loggers. All loggers must inherit the Logger interface (:ref:`logger-interface`).  You can also check out `logger.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/logger.py>`__ for implementation details.
+
+An example can be found in `logging_example.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/logging_example.py>`__.
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/tune/api_docs/logging.rst]: done, good point

File: doc/source/tune/api_docs/schedulers.rst (hunk 3 above)
Comment (richardliaw) [reply doc/source/tune/api_docs/schedulers.rst:6]: added

File: doc/source/tune/api_docs/suggestion.rst (hunk 2 above)
Comment (richardliaw) [reply doc/source/tune/api_docs/suggestion.rst:9]: done

File: doc/source/tune/api_docs/schedulers.rst [hunk 7]
```diff
@@ -1,38 +1,173 @@
...
 
 .. autoclass:: ray.tune.schedulers.PopulationBasedTraining
 
 
+.. _tune-scheduler-bohb:
+
+BOHB (tune.schedulers.HyperBandForBOHB)
+---------------------------------------
+
+This class is a variant of HyperBand that enables the BOHB Algorithm. This implementation is true to the original HyperBand implementation and does not implement pipelining nor straggler mitigation.
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/tune/api_docs/schedulers.rst]: yep done

File: doc/source/tune/api_docs/schedulers.rst [hunk 8]
```diff
@@ -1,38 +1,173 @@
...
 .. autoclass:: ray.tune.schedulers.ASHAScheduler
 
-MedianStoppingRule
-~~~~~~~~~~~~~~~~~~
+.. _tune-original-hyperband:
+
+HyperBand (tune.schedulers.HyperBandScheduler)
+----------------------------------------------
+
+Tune implements the `standard version of HyperBand <https://arxiv.org/abs/1603.06560>`__.
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/tune/api_docs/schedulers.rst]: done

**Review by bllchmbrs — APPROVED** [review]

**Pull request merged**